### ISSUE TITLE
Property doesn't deserialize 

### ISSUE THREAD

**Issue by redmitry** [eclipse-ee4j/yasson#81]
Hello,

I have an issue with a property deserialization:
```
@JsonbProperty("api")
public String getAPIDocumentation() {
    return api;
}
```
Please take a look at the attached maven test project.

Dmitry

**Comment by redmitry** [comment]
[yasson.zip](https://github.com/eclipse/yasson/files/1642613/yasson.zip)

**Comment by moghaddam** [comment]
Hi @redmitry 

Seems it's due to a bug in deserialization that happens in an edge case like the one you provided. 

When we have a pair of getter/setter methods that are not named based on JavaBean convention (as in your case, your property is named 'api' but its getter/setter methods are not getApi and setApi) and then you explicitly define the `@JsonProperty` for getter/setter method providing the exact same name as the underlying field (e.g. `@JsonProperty("api"`) in your case), then the deserialization process will fail.

It can succeed if configuring some naming strategies (e.g. `LOWER_CASE_WITH_UNDERSCORES` or  `UPPER_CASE_WITH_UNDERSCORES`) but not for some other strategies like (`CASE_INSENSITIVE` which is probably the default one).

**Comment by m0mus** [comment]
@moghaddam any progress here?

**Comment by moghaddam** [comment]
Yes, I think I fixed it finally.

**Comment by aguibert** [comment]
For future reference, this was the class in question from inside the original zip:

```java
public class TheBean {
    
    private String api;
    
    @JsonbProperty("api")
    public String getAPIDocumentation() {
        return api;
    }
    
    @JsonbProperty("api")
    public void setAPIDocumentation(String api) {
        this.api = api;
    }
}
```

And here are the attached tests:
```java
    @Test // this one passes
    public void test1() {
        final Jsonb jsonb = JsonbBuilder.create(
                new JsonbConfig().withPropertyNamingStrategy(PropertyNamingStrategy.UPPER_CAMEL_CASE));

        final TheBean bean1 = new TheBean();
        bean1.setAPIDocumentation("REST");

        final String json = jsonb.toJson(bean1);
        final TheBean bean2 = jsonb.fromJson(json, TheBean.class);
        Assert.assertEquals("api properties not equeal", bean1.getAPIDocumentation(), bean2.getAPIDocumentation());
    }
    
    @Test // this one fails because jsonb.toJson(bean1) returns empty json
    public void test2() {
        final Jsonb jsonb = JsonbBuilder.create();
        
        final TheBean bean1 = new TheBean();
        bean1.setAPIDocumentation("REST");

        final String json = jsonb.toJson(bean1);
        final TheBean bean2 = jsonb.fromJson(json, TheBean.class);
        Assert.assertEquals("api properties not equeal", bean1.getAPIDocumentation(), bean2.getAPIDocumentation());
    }
```

**Comment by Degubi** [comment]
I think this got fixed too in #357 because the tests now pass @aguibert

**Comment by aguibert** [comment]
thanks for checking @Degubi, would you like to add the test case in another PR and we can use that PR to close this issue?

**Comment by Degubi** [comment]
Sure @aguibert: #360

**Comment by aguibert** [comment]
Resolved by #357 and a specific test added in #360 by @Degubi.

Closing this issue
